toolchain/check/handle_binding_pattern.cpp

```diff
@@ -128,15 +128,12 @@ static auto HandleAnyBindingPattern(Context& context, Parse::NodeId node_id,
         auto field_type_id = context.GetUnboundElementType(
             class_info.self_type_id, cast_type_id);
         auto field_id = context.AddInst<SemIR::FieldDecl>(
-            binding_id,
-            {.type_id = field_type_id,
-             .name_id = name_id,
-             .index = SemIR::ElementIndex(
-                 context.struct_type_fields_stack().PeekArray().size())});
+            binding_id, {.type_id = field_type_id,
+                         .name_id = name_id,
+                         .index = SemIR::ElementIndex(-1)});
```

Comment:
Please add an `ElementIndex::Invalid` for use here, for consistency with our other invalid index values.
```suggestion
                         .index = SemIR::ElementIndex::Invalid});
```